message_test.go

```diff
@@ -182,6 +182,34 @@ func TestParseMessageError(t *testing.T) {
 	}
 }
 
+func TestMarshalMessageChecksum(t *testing.T) {
+	source := net.ParseIP("2001:db8::10")
+	destination := net.ParseIP("2001:db8::1")
+	mac, _ := net.ParseMAC("86:31:82:05:ce:9a")
```

Comment:
Add error check.  I assume this MAC address comes from some documentation and not a production system?

If it's Wikipedia or similar I'd appreciate a link so it can be verified independently by the reader.

Reply:
Added error check.

The MAC is a random one generated in the locally administered space (0x02 bit is set in the first octet). I don't make it a practice to test code with documentation prefixes for MACs or IPs because that can run into problems if the code happens to do something special *because* it is a documentation prefix (which I've seen happen before). But, if you insist, I can use something like `00-00-5E-00-53-10` from the IANA documentation allocation created [in 2013][1]. Before that, I don't think there was a documentation prefix for MAC addresses.

[1]: https://tools.ietf.org/html/rfc7042#page-7

Reply:
Nah it's fine; I just wanted to make sure you didn't copy this off your own machine or similar.